include/mfmg/dealii_operator.templates.hpp

```diff
@@ -148,6 +148,96 @@ DealIITrilinosMatrixOperator<VectorType>::build_range_vector() const
 
 //-------------------------------------------------------------------------//
 
+dealii::LinearAlgebra::distributed::Vector<double>
+extract_row(dealii::TrilinosWrappers::SparseMatrix const &matrix,
+            dealii::types::global_dof_index global_j)
+{
+  int num_entries = 0;
+  double *values = nullptr;
+  int *local_indices = nullptr;
+  ASSERT(matrix.trilinos_matrix().IndicesAreLocal(), "Indices are not local");
+  auto const local_j =
+      matrix.locally_owned_range_indices().index_within_set(global_j);
+  if (local_j != dealii::numbers::invalid_dof_index)
+  {
+    auto const error_code = matrix.trilinos_matrix().ExtractMyRowView(
+        local_j, num_entries, values, local_indices);
+    ASSERT(error_code == 0,
+           "Non-zero error code (" + std::to_string(error_code) +
+               ") returned by Epetra_CrsMatrix::ExtractMyRowView()");
+  }
+
+  dealii::IndexSet ghost_indices(matrix.locally_owned_domain_indices().size());
+  for (int k = 0; k < num_entries; ++k)
+    ghost_indices.add_index(matrix.trilinos_matrix().GCID(local_indices[k]));
```

Comment:
This creates a "custom" `IndexSet` for this row, which is different than the global `Epetra_Map`, right?